app/modules/entities/components/layouts/works_browser_facets.svelte

```diff
@@ -1,31 +1,39 @@
 <script>
   import SelectDropdown from '#components/select_dropdown.svelte'
-  import { i18n } from '#user/lib/i18n'
-  import { entityProperties, getWorksFacets } from '#entities/components/lib/works_browser_helpers'
-  import Spinner from '#components/spinner.svelte'
+  import { entityProperties, getWorksFacets, getFacetsSelectors } from '#entities/components/lib/works_browser_helpers'
   import { getContext } from 'svelte'
+  import { i18n, I18n } from '#user/lib/i18n'
 
   export let works, facets, facetsSelectors, facetsSelectedValues, flash
 
   const layoutContext = getContext('layout-context')
+  const facetsObj = getWorksFacets({ works, context: layoutContext })
 
-  const waitForFacets = getWorksFacets({ works, context: layoutContext })
-    .then(res => ({ facets, facetsSelectedValues, facetsSelectors } = res))
-    .catch(err => flash = err)
-</script>
+  let loadingMessage
+  const displayLoadingInfoMessage = async facetsObj => {
+    const urisSize = facetsObj.valuesUris.length
+    loadingMessage = I18n('loading_may_take_a_while', { smart_count: urisSize })
+  }
```

Comment:
Nothing async here, and see below for why `loadingMessage` could be a `const`. Also, if the message includes `urisCount`, this will probably not be a very generalist message, so having a more specific key (such as `loading_facets_may_take_a_while`) could be helpful:
```suggestion
  const urisCount = facetsObj.valuesUris.length
  const loadingMessage = I18n('loading_facets_may_take_a_while', { smart_count: urisCount })
```

Reply:
done in 98884209d